File: example/src/main/java/net/droidlabs/viking/example/dialog/mvvm/DialogMvvmFragment.java
package net.droidlabs.viking.example.dialog.mvvm;

import net.tango.viking.annotations.AutoModule;
import net.droidlabs.viking.example.R;
import net.droidlabs.viking.example.TestViewModel2;
import net.droidlabs.viking.example.databinding.FragmentVikingDialogBinding;
import net.droidlabs.viking.mvvm.DialogFragmentView;

@AutoModule
public class DialogMvvmFragment extends DialogFragmentView<TestViewModel2, FragmentVikingDialogBinding> {
  public static DialogMvvmFragment newInstance() {
    return new DialogMvvmFragment();
  }

  public DialogMvvmFragment() {
    super(R.layout.fragment_viking_dialog, TestViewModel2.class);
  }

  @Override
  protected void bind(FragmentVikingDialogBinding binding) {

  }
}

File: example/src/main/java/net/droidlabs/viking/example/dialog/mvvm/DialogMvvmFragment.kt
package net.droidlabs.viking.example.dialog.mvvm

import net.droidlabs.viking.annotations.AutoModule
import net.droidlabs.viking.example.R
import net.droidlabs.viking.example.TestViewModel2
import net.droidlabs.viking.example.databinding.FragmentVikingDialogBinding
import net.droidlabs.viking.mvvm.DialogFragmentView

@AutoModule
class DialogMvvmFragment : DialogFragmentView<TestViewModel2, FragmentVikingDialogBinding>(
    layoutIdRes = R.layout.fragment_viking_dialog,
    viewModelClass = TestViewModel2::class.java
) {
  override fun bind(binding: FragmentVikingDialogBinding) {}

  companion object {
    fun newInstance(): DialogMvvmFragment {
      return DialogMvvmFragment()
    }
  }
}